nodes/src/nodes/router_llm/router.py

```
@@ -0,0 +1,337 @@
+        if self.budget_limit > 0 and current_cost >= self.budget_limit * 0.8:
+            info = _get_model_info('claude-sonnet')
+            info['reason'] = f'approaching budget limit (spent ${current_cost:.4f} of ${self.budget_limit:.4f}), routed to balanced model'
+            self.record_cost(_tier_estimated_cost.get(info['tier'], 0.01))
+            return info
+
+        info = _get_model_info(self.primary_model)
+        info['reason'] = f'within budget (spent ${current_cost:.4f} of ${self.budget_limit:.4f}), using primary model'
+        self.record_cost(_tier_estimated_cost.get(info['tier'], 0.01))
+        return info
+
+    def _route_latency(self, text: str) -> Dict[str, Any]:
```

Ruff (0.15.9)

[warning] Unused method argument: `text`

(ARG002)